.github/workflows/build.yml

```diff
@@ -60,5 +60,5 @@ jobs:
 
           set -x
           cargo fmt --check
-          cargo clippy
+          cargo clippy -- --deny "warnings"
```

Comment:
This will bring the interactive commands (as described in the markdown docs) out of sync with the CI commands. Perhaps leave this out of this pull request for now and lets think about these commands separately? It would be nice if CI did the same thing as what we tell the users to do interactively.

Reply:
Interactive commands always tell you both errors and warnings, but in CI we want warnings to make the command return an error.

Using -Dwarnings is a really bad experience because it's really confusing whether something is a type error or just a warning (which rust has a LOT of). Even simple things like unused imports would error, making it really confusing to navigate the code.